pages/shop/[pid].js

```diff
@@ -1,29 +1,58 @@
 import React, { useContext, useState } from 'react';
-import { useRouter } from 'next/router';
 
 import Head from 'next/head';
 import Link from 'next/link';
 
-import Button from '@material-ui/core/Button';
-
 import { CartContext } from '../../src/contexts/CartContext';
+import { getProduct } from '../../lib/api';
 
-import { getProducts, getProduct } from '../../lib/api';
+import Grid from '@material-ui/core/Grid';
+import Button from '@material-ui/core/Button';
 import TextField from '@material-ui/core/TextField';
+import MenuItem from '@material-ui/core/MenuItem';
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+    margin: '1rem',
+    '& .MuiTextField-root': {
+      margin: theme.spacing(1),
+      width: '10rem',
+    },
+  },
+}));
 
-export default function Product({ productData }) {
+export default function Product({ productData, variationData }) {
   const { addProduct, cartItems, increase } = useContext(CartContext);
+  const classes = useStyles();
 
   const [quantity, setQuantity] = useState(0);
+  const [size, setSize] = useState('A2');
+
+  let startPrice = variationData.filter((item) => item.attributes[0].option === 'A2')[0].price;
+  const [price, setPrice] = useState(startPrice);
 
-  console.log('rendering');
   const isInCart = (product) => {
     return !!cartItems.find((item) => item.id === product.id);
   };
 
   const addToCart = (productData) => {
-    productData.quantity = parseInt(quantity);
-    addProduct(productData);
+    const selectedVariation = variationData.filter((item) => item.attributes[0].option === size)[0];
+
+    selectedVariation.quantity = quantity;
+    selectedVariation.product_id = productData.id;
```

Comment:
FYI these lines will throw if it couldn't find a `selectedVariation`. Just keep that in mind.

Reply:
Would you wrap this in an `if` to check that the selected variation was found?

Reply:
I would do something like

```
const selectedVariation = variationData.filter((item) => item.attributes[0].option === size)[0] || {}; 
```

That'll provide a fallback empty object for `selectedVariation` if it couldn't find one.